openupgrade_scripts/scripts/hr_holidays/15.0.1.5/pre-migration.py

```diff
@@ -238,6 +238,26 @@ def _create_column_hr_leave_holiday_allocation_id(env):
     )
 
 
+def _map_hr_leave_allocation_temp_date_to(env):
+    """
+    set null the date_to column to fill `holiday_allocation_id` column
+    after re-update `date_to` column at post-migrate 
+    """
+    openupgrade.logged_query(
+        env.cr,
+        """
+        ALTER TABLE hr_leave_allocation
+        ADD COLUMN IF NOT EXISTS temp_date_to DATE""",
+    )
```

Comment:
@vietphuong10 như này cho chính tắc nhé, sẽ tạo ra column dạng `openupgrade_legecy_15_0_date_to`
```suggestion
    openupgrade.copy_columns(
        env.cr,
        {
            "hr_leave_allocation": [
                ("date_to", None, None),
            ],
        },
    )
```